generator/src/main/java/com/faforever/neroxis/generator/texture/BasicTextureGenerator.java

```diff
@@ -90,14 +109,16 @@ public void initialize(SCMap map, long seed, GeneratorParameters generatorParame
         steepHillsTexture = new FloatMask(1, random.nextLong(), symmetrySettings, "steepHillsTexture", true);
         rockTexture = new FloatMask(1, random.nextLong(), symmetrySettings, "rockTexture", true);
         accentRockTexture = new FloatMask(1, random.nextLong(), symmetrySettings, "accentRockTexture", true);
+        terrainType = new IntegerMask(1, random.nextLong(), symmetrySettings, "terrainType", true);
     }
 
     @Override
     public void setTextures() {
-        Pipeline.await(texturesLowMask, texturesHighMask);
+        Pipeline.await(texturesLowMask, texturesHighMask, terrainType);
         DebugUtil.timedRun("com.faforever.neroxis.map.generator", "generateTextures", () -> {
             map.setTextureMasksScaled(map.getTextureMasksLow(), texturesLowMask.getFinalMask());
             map.setTextureMasksScaled(map.getTextureMasksHigh(), texturesHighMask.getFinalMask());
+            map.setTerrainTypeScaled(map.getTerrainType(), terrainType);
```

Comment:
Do I need getFinalMask() here? Not really sure what it does

Reply:
Ah yes it would be good practice. getFinalMask just guarantees all processing for that mask is finished